pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -195,86 +204,118 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 					// We return earlier for all error cases so we don't need
 					// to init the local endpoint in non-error cases.
 					needInit = false
-				}
+					return nil
+				} else {
```

Comment:
We typically don't write:
```
     [...]
     return
 } else {
     [...]
 ````
 
 But instead:
 ````
     [...]
     return
 }
[...]
````

Reply:
Done :) Also updated commit msg to be slightly clearer.